pages/docs/concepts/asyncapi-document/extending-specification.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Extending Specification
+weight: 220
+---
+
+Extending the AsyncAPI specification is a technique that allows developers to include domain-specific or use-case-specific information not supported by the base specification. This extension capability provides customization, enabling APIs to accommodate unique details that would not otherwise fit within the confines of the standard AsyncAPI specification.
+
+The benefits of extending the AsyncAPI specification include enhanced specificity — delivering a clear encoding of necessary API details to clients, and consistency — allowing reusable components across the API. However, extensions should be used sparingly. Since they go beyond the Specification, they may not be universally understood.
+
+## AsyncAPI Specification Extensions
+
+The AsyncAPI Specification allows the addition of custom properties through patterned fields prefixed with `x-`. This feature enables the accommodation of specific functionalities not initially included in the AsyncAPI specification.
+
+The `x-` prefix is used to define custom properties. These properties are user-defined and won't conflict with future specification versions because any property starting with `x-` is reserved for user definitions, and the AsyncAPI organization will not use this prefix for any of their future standard properties.
+
+```mermaid
+flowchart TB
+    subgraph "AsyncAPI Specification"
+        A[Specification] --> B[Custom Properties]
+    end
+    subgraph "Custom Properties"
+        B --> C[Patterned fields prefixed with x-]
+        B --> D[Reserved for user definitions]
+        B --> E[Support any valid JSON format value]
+    end
+    subgraph "Usage"
+        C --> F[Accommodate specific functionalities]
+        D --> G[Avoid conflict with future versions]
+        E --> H[May or may not be supported by tools]
+    end
+    subgraph "Tooling"
+        H --> I[Internal tools]
+        H --> J[Open-sourced tools]
+        I --> K[Extended support may be added]
+        J --> L[Extended support may be added]
+    end
+```
```

Comment:
TBH, I found this more confusing than helping.

What exactly is this diagram supposed to show and inform the reader about?

Reply:
I have simplified the diagram.